### DISCUSSION TITLE
H2 does not receieve midi events, although PC does

### DISCUSSION THREAD

**Discussion by kitschen** [hydrogen-music/hydrogen#1112] [Hydrogen on Windows]
Hello
My PC (Win10) gets MIDI events through an external USB sound card (Behringer UMC204HD), but H2 does not receive any of those MIDI events. The audio output through the card works fine.

The midi messages correctly arrive on the PC. They can be seen in Midi-Ox  and other Midi enabled programs.
Screenshot from config in Midi-Ox
![image](https://user-images.githubusercontent.com/3657877/105455382-ad20a100-5c83-11eb-960f-99b8d90f0aec.png)

H2 config shows only the PortMidi driver. Should it show others?
Must I configure PortMidi elsewhere to get the events?
Do you have other ideas what I can try to get the midi events into H2?

To confirm that no midi arrives, I have tried the H2 "learn" function (to assign to actions). The remains showing the "Waiting for midi input..." message.

![image](https://user-images.githubusercontent.com/3657877/105455792-641d1c80-5c84-11eb-86c3-81728e06747e.png)

**Comment by kitschen** [comment]
The logs collected by `-VInfo` show some issue with `Unknown midi input from preferences [ALSA]`
```
(I) Preferences::loadPreferences Loading preferences file (USER) [C:/Users/Philipp/.hydrogen/hydrogen.conf]
(W) LocalFileMng::readXmlString         using default value : '' for node 'lastPlaylistFilename'
(W) LocalFileMng::readXmlString         using default value : '' for node 'defaulteditor'
(E) PreferencesDialog::PreferencesDialog Unknown midi input from preferences [ALSA]
(I) PreferencesDialog::~PreferencesDialog ~PREFERENCES_DIALOG
```
Here's the relevant part in the config
```
  <midi_driver>
   <driverName>ALSA</driverName>
   <port_name>None</port_name>
   <output_port_name>None</output_port_name>
   <channel_filter>-1</channel_filter>
   <ignore_note_off>false</ignore_note_off>
   <enable_midi_feedback>false</enable_midi_feedback>
   <discard_note_after_action>true</discard_note_after_action>
   <fixed_mapping>false</fixed_mapping>
  </midi_driver>
```

Full config file attached.
[hydrogen.txt](https://github.com/hydrogen-music/hydrogen/files/5854221/hydrogen.txt)

**Comment by theGreatWhiteShark** [comment]
Might be related to #531

**Comment by cme** [comment]
PortAudio/PortMidi is the only audio API supported on Windows. It's possible this might also be related somewhat to #611 -- we have no way to configure PortAudio at all.

I don't really have a Windows machine either, but had been considering looking at the APIs and working on #611 since it's workable without actually having a Windows machine. It's possible there's a common solution.

**Reply by kitschen** [reply]
If you need additional info or testing of a beta, do not hesitate to get in touch.

**Comment by dfkettle** [comment]
Any update on this issue? I'm having trouble sending MIDI data from Hydrogen to a drum machine (an Arturia Drumbrute). I'm able to send MIDI data to it from other software (Python, for example), but not Hydrogen. But I'm new to Hydrogen, it may be something I'm doing wrong.

**Reply by theGreatWhiteShark** [reply]
Did you already took a look at the MIDI Out Settings section of the [manual](http://hydrogen-music.org/documentation/manual/manual_en_single_page.html#chpt.instrument_editor.general)?

About the general issue addressed here: unfortunately, there is nobody in our team right now who is either working on (or has the resources to look into) Windows or MIDI.

**Reply by dfkettle** [reply]
> Did you already took a look at the MIDI Out Settings section of the [manual](http://hydrogen-music.org/documentation/manual/manual_en_single_page.html#chpt.instrument_editor.general)?
> 
Yes, I did. I tried setting the channel number to different values and monitored it in MIDI-OX, but didn't see any data being sent. I'm guessing that "Off" means don't send MIDI data, correct?

Are there any other features that don't work in the Windows version, apart from MIDI and effects? It might save everyone, including the developers, some time if you listed the features supported in each version somewhere on the web site.

**Reply by theGreatWhiteShark** [reply]
> Yes, I did. I tried setting the channel number to different values and monitored it in MIDI-OX, but didn't see any data being sent. I'm guessing that "Off" means don't send MIDI data, correct?

Yes. But you also need to set a MIDI driver / output combination in the Preferences > MIDI System tab.

> Are there any other features that don't work in the Windows version, apart from MIDI and effects? It might save everyone, including the developers, some time if you listed the features supported in each version somewhere on the web site.

I also thought about it. But since I have no Windows machine I can not tell. If you want, you can do so yourself in our [wiki](https://github.com/hydrogen-music/hydrogen/wiki).

**Comment by nunosga** [comment]
Hi,
I'm trying to use a midi keyboard (Korg Microkey) to play / record patterns in Hydrogen, but, just like user Kitchen,  i can't get Hydrogen to work with the keyboard. (the Korg works fine with the studio one DAW).
I'm not a programmer (only made a few batch (MS-Dos) code lines in the day).  What should i do?
